game/scripts/core/player_interactor.gd

```diff
@@ -0,0 +1,36 @@
+extends Interactor
+
+@export var player: Player
+
+var closest_interactable: Interactable
+
+func _ready():
+	controller = player
+	print(controller)
+	
+func _physics_process(delta: float) -> void:
+	update_closest_interactable()
```

Comment:
I wonder if there's any ways that we can optimize this to not run every render cycle. probably fine for now though.